src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -222,6 +222,12 @@ private Instrumenter Instrumenter
 
         private BoundExpression? VisitExpressionImpl(BoundExpression node)
         {
+            if (node is BoundNameOfOperator { WasCompilerGenerated: false, Syntax: InvocationExpressionSyntax { Expression: IdentifierNameSyntax { Location: var nameofPseudoCallLocation } } }
```

Comment:
Interesting. Besides `nameof` are there other expressions that use invocation syntax but aren't interceptable? 
Looking at `BindInvocationExpression`, maybe `__arglist()` would fit that category too.
Also, what should we do about dynamic, delegate and function pointer invocations? I assume none of them can be intercepted.

nit: do we need the `WasCompilerGenerated: false` check?

Reply:
Good test cases to add. `__arglist()` is actually not invocation syntax unlike `nameof`. But we'll maybe want to add tests for anything that looks like it (`__makeref`, `__refvalue`, `__reftype` also come to mind)

Reply:
re: delegates. See existing test case `InterceptableDelegateInvocation`. It's currently only blocked by the fact that delegate Invoke is not marked with `InterceptableAttribute`. We are likely to drop `InterceptableAttribute` from the design, which would open that scenario up.

My feeling currently is that if you want to intercept a call to a delegate, there's no reason for us to stop you--you must have decided what that delegate call means, and you have to take the delegate the user is invoking as an argument as well, so I could imagine wanting to intercept teh call simply to wrap it and do something before/after calling the actual delegate.

Dynamic and function pointers I might have to experiment with in order to form an opinion.

Reply:
I added delegate, dynamic and function pointers to the test plan as a reminder to address down the line. I lean towards disallowing all three.